### PR TITLE
Add phase page + story page

### PR DESCRIPTION
**Pull request by duanxianbo**
[(task:20571)](https://staging.teamyapp.com/teams/107/tasks/20571)

### CODE REVIEW

File: src/components/internal/SelectOwnersPopupComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import { MaterialIconUI } from '@lib/ui/MaterialIcon';
+import { BoundingBoxFinder } from '@lib/ui/position';
+
+import { Deps } from '@core/dep/deps';
+import { User } from '@core/entity/user';
+import { GraphSource } from '@core/storage/graph/graphSource';
+import { StateSyncer } from '@core/storage/syncer/stateSyncer';
+
+import styles from './SelectOwnersPopupComponent.module.scss';
+import { SelectUserListUI } from './SelectUserList';
```
Comment (naturetime10) [inline src/components/internal/SelectOwnersPopupComponent.tsx]:
```suggestion
import { SelectUserListUI } from './SelectUserList';
import styles from './SelectOwnersPopupComponent.module.scss';
```

File: src/components/internal/SelectOwnersPopupComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+    constructor(props: Props) {
+        super(props);
+        this.graphSource = props.deps.graphSource;
+        this.stateSyncer = props.deps.stateSyncer;
+        this.state = {
+            showPotentialOwnersList: false,
+            OwnersListVisible: false,
+            OwnersListLeft: 0,
+            OwnersListTop: 0,
```
Comment (naturetime10) [inline src/components/internal/SelectOwnersPopupComponent.tsx]:
```suggestion
            ownersListVisible: false,
            ownersListLeft: 0,
            ownersListTop: 0,
```

File: src/components/internal/SelectOwnersPopupComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    findOwnersAlignTargetBoundBox?: BoundingBoxFinder;
+    onClose?: () => void;
+    onSelectOwner?: (OwnerId?: number) => void;
+}
+
+interface State {
+    showPotentialOwnersList: boolean;
+    OwnersListVisible: boolean;
+    OwnersListLeft: number;
+    OwnersListTop: number;
```
Comment (naturetime10) [inline src/components/internal/SelectOwnersPopupComponent.tsx]:
```suggestion
    ownersListVisible: boolean;
    ownersListLeft: number;
    ownersListTop: number;
```

File: src/components/internal/SelectOwnersPopupComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    potentialOwners: User[];
+}
+
+export class SelectOwnersPopupComponent extends Component<Props, State> {
+    private readonly graphSource: GraphSource;
+    private readonly stateSyncer: StateSyncer;
+    private readonly OwnersPopupRef = createRef<HTMLDivElement>();
```
Comment (naturetime10) [inline src/components/internal/SelectOwnersPopupComponent.tsx]:
```suggestion
    private readonly ownersPopupRef = createRef<HTMLDivElement>();
```

File: src/components/internal/SelectOwnersPopupComponent.tsx [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    render() {
+        return (
+            this.state.showPotentialOwnersList && (
+                <div
+                    ref={this.OwnersPopupRef}
+                    className={styles.SelectOwnersPopup}
+                    style={{
+                        left: this.state.OwnersListLeft,
+                        top: this.state.OwnersListTop,
+                        visibility: this.state.OwnersListVisible
```
Comment (naturetime10) [inline src/components/internal/SelectOwnersPopupComponent.tsx]:
```suggestion
                        visibility: this.state.ownersListVisible
```

File: src/components/internal/SelectOwnersPopupComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+    render() {
+        return (
+            this.state.showPotentialOwnersList && (
+                <div
+                    ref={this.OwnersPopupRef}
+                    className={styles.SelectOwnersPopup}
+                    style={{
+                        left: this.state.OwnersListLeft,
+                        top: this.state.OwnersListTop,
```
Comment (naturetime10) [inline src/components/internal/SelectOwnersPopupComponent.tsx]:
```suggestion
                        left: this.state.ownersListLeft,
                        top: this.state.ownersListTop,
```

File: src/components/internal/SelectOwnersPopupComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+        this.setState({
+            potentialOwners: teamMembers.map(
+                (teamMember) => teamMember.user,
+            ),
+            showPotentialOwnersList: true,
+        });
+
+        setTimeout(() => {
+            this.setState({
+                OwnersListVisible: true,
```
Comment (naturetime10) [inline src/components/internal/SelectOwnersPopupComponent.tsx]:
```suggestion
                ownersListVisible: true,
```

File: src/components/internal/SelectOwnersPopupComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+            });
+
+            this.OwnersPopupRef.current?.focus();
```
Comment (naturetime10) [inline src/components/internal/SelectOwnersPopupComponent.tsx]:
```suggestion
            this.ownersPopupRef.current?.focus();
```

File: src/components/internal/SelectOwnersPopupComponent.tsx [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+            return;
+        }
+
+        if (!this.props.findOwnersAlignTargetBoundBox) {
+            return;
+        }
+
+        const boundingBox = this.props.findOwnersAlignTargetBoundBox();
+        const width = this.OwnersPopupRef.current?.offsetWidth || 0;
+        const height = this.OwnersPopupRef.current?.offsetHeight || 0;
```
Comment (naturetime10) [inline src/components/internal/SelectOwnersPopupComponent.tsx]:
```suggestion
        const width = this.ownersPopupRef.current?.offsetWidth || 0;
        const height = this.ownersPopupRef.current?.offsetHeight || 0;
```

File: src/components/pages/Planning/InlineCreateStory.component.tsx [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+    onCreateStory?: (story: CreateStoryInput) => void;
+    onDiscardNewStory?: () => void;
+}
+
+export const InlineCreateStory: FC<Props> = (props) => {
+    const [story, setStory] = useState<CreateStoryInput>({
+        name: '',
+        ownerId: props.initialOwnerId || 0,
+    });
+    // let getBoundingClientRect: (() => DOMRect) | undefined = undefined;
```
Comment (duanxianbo) [inline src/components/pages/Planning/InlineCreateStory.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/InlineCreateStory.component.module.scss [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+        .Action {
+            display: flex;
+            width: $actionIconSize;
+            height: $actionIconSize;
+            padding: 4px;
+            border-radius: 4px;
+            align-items: center;
+            --icon-box-size: #{$actionIconSize};
+            justify-content: center;
+            --icon-size: #{$actionIconSize};
```
Comment (naturetime10) [inline src/components/pages/Planning/InlineCreateStory.component.module.scss]:
```suggestion
            justify-content: center;
            --icon-box-size: #{$actionIconSize};
            --icon-size: #{$actionIconSize};
```

File: src/components/pages/Planning/InlineCreateStory.component.tsx (hunk 10 above)
Comment (naturetime10) [inline src/components/pages/Planning/InlineCreateStory.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/Phase.component.module.scss [hunk 12]
```diff
@@ -0,0 +1,131 @@
+.Phase {
+    display: flex;
+    height: 100%;
+    .PhaseDetail {
```
Comment (naturetime10) [inline src/components/pages/Planning/Phase.component.module.scss]:
```suggestion
    height: 100%;
    
    .PhaseDetail {
```

File: src/components/pages/Planning/Phase.component.tsx [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+
+    const onPhaseStorySelect = (storyIdNumber: number) => {
+        phaseStorySelectorRef.current.selectItem(storyIdNumber.toString());
+    };
+
+    const onClickStoryDetailHandler = async (storyId: number) => {
+        // await props.deps.stateSyncer.pullCurrentTeamTaskPreview({
+        //     isPlanned: false,
+        // });
+
```
Comment (duanxianbo) [inline src/components/pages/Planning/Phase.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/Project.component.tsx [hunk 14]
```diff
@@ -8,6 +8,7 @@ import {
     useRef,
     useState,
 } from 'react';
+
```
Comment (duanxianbo) [inline src/components/pages/Planning/Project.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/Phase.component.tsx [hunk 15]
```diff
@@ -0,0 +1,275 @@
...
+import { Deps } from '@core/dep/deps';
+import { CreateStoryInput } from '@core/entity/input';
+import { Phase } from '@core/entity/phase';
+import { Project } from '@core/entity/project';
+import { phaseStoryPath } from '@core/routing/routes';
+
+import { InlineCreateStory } from './InlineCreateStory.component';
+import styles from './Phase.component.module.scss';
+import { StoryInlineItemComponent } from './StoryInlineItem.component';
```
Comment (naturetime10) [inline src/components/pages/Planning/Phase.component.tsx]:
```suggestion
import { StoryInlineItemComponent } from './StoryInlineItem.component';
import styles from './Phase.component.module.scss';
```

File: src/components/pages/Planning/Project.component.tsx [hunk 16]
```diff
@@ -215,16 +217,17 @@ export function ProjectComponent(props: Props) {
...
             </div>
         );
     };
     const onStoryDetailClickHandler = async (storyId: number) => {
-        await props.deps.stateSyncer.pullCurrentTeamTaskPreview({
-            isPlanned: false,
-        });
+        // await props.deps.stateSyncer.pullCurrentTeamTaskPreview({
+        //     isPlanned: false,
+        // });
```
Comment (duanxianbo) [inline src/components/pages/Planning/Project.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/Phase.component.tsx (hunk 13 above)
Comment (naturetime10) [inline src/components/pages/Planning/Phase.component.tsx]: Is this still needed?

File: src/components/pages/Planning/Phase.component.tsx [hunk 17]
```diff
@@ -0,0 +1,275 @@
...
+        //     story,
+        // );
+        setOnEditingNewStoryInPhase(false);
+    };
+
+    const onCreateStoryInBacklog = (story: CreateStoryInput) => {
+        // props.deps.stateSyncer.createStory(
+        //     props.deps.localStore.getState().currTeamId!,
+        //     story,
+        // );
```
Comment (naturetime10) [inline src/components/pages/Planning/Phase.component.tsx:169]: Will this be implemented?

File: src/components/pages/Planning/StoryInlineItem.component.tsx [hunk 18]
```diff
@@ -0,0 +1,233 @@
...
+        };
+
+        const handleStoryClick = (event: MouseEvent) => {
+            event.stopPropagation();
+            props.onStorySelectHandler(props.story.id);
+        };
+
+        const onAssignOwnerClick = (event: MouseEvent<HTMLDivElement>) => {
+            const element = event.currentTarget as HTMLDivElement;
+            // getBoundingClientRect = element.getBoundingClientRect.bind(element);
```
Comment (duanxianbo) [inline src/components/pages/Planning/StoryInlineItem.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/Phase.component.tsx [hunk 19]
```diff
@@ -0,0 +1,275 @@
...
+
+    const onAddStoryInBacklogClick = () => {
+        setOnEditingNewStoryInBacklog(true);
+    };
+
+    const onCreateStoryInPhase = (story: CreateStoryInput) => {
+        // props.deps.stateSyncer.createStory(
+        //     props.deps.localStore.getState().currTeamId!,
+        //     story,
+        // );
```
Comment (naturetime10) [inline src/components/pages/Planning/Phase.component.tsx:161]: Will this be implemented?

File: src/components/pages/Planning/Planning.component.tsx [hunk 20]
```diff
@@ -20,9 +20,11 @@ import {
 } from '@core/routing/routes';
 import { toInt } from '@core/storage/states/parser';
 
+import { PhaseComponent } from './Phase.component';
+import styles from './Planning.component.module.scss';
```
Comment (naturetime10) [inline src/components/pages/Planning/Planning.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/Planning.component.tsx [hunk 21]
```diff
@@ -20,9 +20,11 @@ import {
...
 import { ProjectComponent } from './Project.component';
 import { ProjectsComponent } from './Projects.component';
-import styles from './Planning.component.module.scss';
+import { StoryComponent } from './Story.component';
```
Comment (naturetime10) [inline src/components/pages/Planning/Planning.component.tsx:26]:
```suggestion
import { StoryComponent } from './Story.component';
import styles from './Planning.component.module.scss';
```

File: src/components/pages/Planning/Project.component.tsx [hunk 22]
```diff
@@ -18,12 +19,14 @@ import { Phase } from '@core/entity/phase';
...
 import { phasePath, projectStoryPath } from '@core/routing/routes';
 
+import { InlineCreateStory } from './InlineCreateStory.component';
+import styles from './Project.component.module.scss';
+import { StoryInlineItemComponent } from './StoryInlineItem.component';
 import {
     Row,
     TimelineComponent,
     TimelineComponentHandle,
 } from './Timeline.component';
```
Comment (naturetime10) [inline src/components/pages/Planning/Project.component.tsx]:
```suggestion
import { StoryInlineItemComponent } from './StoryInlineItem.component';
import {
    Row,
    TimelineComponent,
    TimelineComponentHandle,
} from './Timeline.component';
import styles from './Project.component.module.scss';
```

File: src/components/pages/Planning/Project.component.tsx (hunk 16 above)
Comment (naturetime10) [inline src/components/pages/Planning/Project.component.tsx]: Is this still needed?

File: src/components/pages/Planning/Story.component.module.scss [hunk 23]
```diff
@@ -0,0 +1,129 @@
...
+        flex: 1;
+        flex-direction: column;
+
+        .Header {
+            font-size: 16px;
+            font-weight: bold;
+            display: flex;
+            padding: 16px;
+            align-items: center;
+            .Information {
```
Comment (naturetime10) [inline src/components/pages/Planning/Story.component.module.scss]:
```suggestion
            align-items: center;
            
            .Information {
```

File: src/components/pages/Planning/Story.component.module.scss [hunk 24]
```diff
@@ -0,0 +1,129 @@
...
+                display: flex;
+                flex-direction: column;
+                gap: 4px;
+                .Description {
```
Comment (naturetime10) [inline src/components/pages/Planning/Story.component.module.scss]:
```suggestion
                gap: 4px;
                
                .Description {
```

File: src/components/pages/Planning/Story.component.module.scss [hunk 25]
```diff
@@ -0,0 +1,129 @@
...
+                    font-size: 12px;
+                    font-weight: bold;
+                }
+            }
+
+            .Actions {
+                display: flex;
+                gap: 8px;
+                margin-left: auto;
+                .Add {
```
Comment (naturetime10) [inline src/components/pages/Planning/Story.component.module.scss]:
```suggestion
                margin-left: auto;
                
                .Add {
```

File: src/components/pages/Planning/Story.component.module.scss [hunk 26]
```diff
@@ -0,0 +1,129 @@
...
+            }
+        }
+    }
+
+    .TasksSection {
+        display: flex;
+        flex-wrap: wrap;
+        row-gap: 16px;
+        column-gap: 8px;
+        .InlineTask {
```
Comment (naturetime10) [inline src/components/pages/Planning/Story.component.module.scss]:
```suggestion
        column-gap: 8px;
        
        .InlineTask {
```

File: src/components/pages/Planning/Story.component.tsx [hunk 27]
```diff
@@ -0,0 +1,343 @@
...
+                console.log('[PlanningComponent] waiting for state changes');
+                const hasChanged = await stateChangeChan!.pop();
+                if (hasChanged === undefined) {
+                    // check undefined instead of falsy because
+                    // a falsy data could be valid data per channel's concern.
+                    return;
+                }
+
+                if (unplannedTasksRef.current) {
+
```
Comment (naturetime10) [inline src/components/pages/Planning/Story.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/Story.component.tsx [hunk 28]
```diff
@@ -0,0 +1,343 @@
...
+                    let tasks =
+                        props.deps.graphSource.currentTeam()?.tasks || [];
+                    
+                    
+                    console.log("tasks+++++", JSON.stringify(tasks, null, 2))
```
Comment (naturetime10) [inline src/components/pages/Planning/Story.component.tsx]:
```suggestion
```

File: src/components/pages/Planning/Story.component.tsx [hunk 29]
```diff
@@ -0,0 +1,343 @@
...
+                        tasks = tasks
```
Comment (naturetime10) [inline src/components/pages/Planning/Story.component.tsx]:
```suggestion
                    tasks = tasks
```

File: src/components/pages/Planning/StoryInlineItem.component.module.scss [hunk 30]
```diff
@@ -0,0 +1,200 @@
...
+        padding: $edgeSpace;
+        border: 4px solid transparent;
+        border-radius: 6px;
+        background-color: white;
+        word-break: break-word;
+
+        &.Selected {
+           outline: 3px solid #ffc107;
+        }
+
```
Comment (naturetime10) [inline src/components/pages/Planning/StoryInlineItem.component.module.scss:29]:
```suggestion
```

File: src/components/pages/Planning/StoryInlineItem.component.module.scss [hunk 31]
```diff
@@ -0,0 +1,200 @@
...
+                    }
+
+                    &.COMPLETED {
+                        background-color: #26882b;
+                    }
+                }
+
+                .Badge {
+                    color: white;
+                    &.TotalTasks {
```
Comment (naturetime10) [inline src/components/pages/Planning/StoryInlineItem.component.module.scss:89]:
```suggestion

                    &.TotalTasks {
```

File: src/components/pages/Planning/StoryInlineItem.component.tsx [hunk 32]
```diff
@@ -0,0 +1,233 @@
...
+import { Story, StoryStatus } from '@core/entity/story';
+
+import { UserProfileUI } from '../../internal/UserProfile';
+import styles from './StoryInlineItem.component.module.scss';
+
+interface StoryInlineItemComponentProps {
+    story: Story;
+    deps: Deps;
+    isSelected?: boolean;
+    onClickStoryDetailHandler: (storyId: number) => void;
```
Comment (naturetime10) [inline src/components/pages/Planning/StoryInlineItem.component.tsx]:
```suggestion
    onStoryDetailClickHandler: (storyId: number) => void;
```

File: src/components/pages/Planning/StoryInlineItem.component.tsx (hunk 18 above)
Comment (naturetime10) [inline src/components/pages/Planning/StoryInlineItem.component.tsx]:
```suggestion
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: src/components/pages/Planning/Phase.component.tsx [hunk 33]
```diff
@@ -136,10 +136,6 @@ export function PhaseComponent(props: Props) {
     };
 
     const onClickStoryDetailHandler = async (storyId: number) => {
```
Comment (naturetime10) [inline src/components/pages/Planning/Phase.component.tsx]:
```suggestion
    const onStoryDetailClickHandler = async (storyId: number) => {
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: src/components/pages/Planning/Phase.component.tsx (hunk 17 above)
Comment (duanxianbo) [reply src/components/pages/Planning/Phase.component.tsx:169]: yes

File: src/components/pages/Planning/Phase.component.tsx (hunk 19 above)
Comment (duanxianbo) [reply src/components/pages/Planning/Phase.component.tsx:161]: yes

File: src/components/pages/Planning/Phase.component.tsx (hunk 13 above)
Comment (duanxianbo) [reply src/components/pages/Planning/Phase.component.tsx]: removed

File: src/components/pages/Planning/Project.component.tsx (hunk 16 above)
Comment (duanxianbo) [reply src/components/pages/Planning/Project.component.tsx]: removed

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**